test/test_ldap_connection.rb

```diff
@@ -47,6 +47,19 @@ def test_list_of_hosts_with_all_hosts_failure
     end
   end
 
+  def test_result_for_connection_failed_is_set
+    flexmock(TCPSocket).should_receive(:new).and_raise(Errno::ECONNREFUSED)
+
+    ldap_client = Net::LDAP.new(host: '127.0.0.1', port: 12345)
+
+    assert_raise Net::LDAP::ConnectionRefusedError do
```

Comment:
I'm not not a reqular user of minitest. If there's a better approach to handle errors, I'm open to fix this. `begin; rescue; end` didn't work for that use case.